### DISCUSSION TITLE
Cartesi smart contracts on akash/ cosmos

### DISCUSSION THREAD

**Discussion by 31trainman** [akash-network/community#496] [Integrations]
What are the benefits of bringing cartesi smart contracts to Akash network ecosystem?

Integrating Cartesi smart contracts into the Akash network ecosystem could enhance decentralized application (dApp) development by leveraging Cartesi's off-chain computation capabilities. This may result in improved scalability, reduced transaction costs, and increased efficiency, offering developers a more versatile and powerful environment for creating decentralized applications on the Akash network.

How much akt will it cost to bring cartesi smart contracts to Akash network?

The cost of bringing Cartesi smart contracts to the Akash network would depend on various factors, including the specific terms of any agreements between the parties involved, market conditions, and the resources required for integration. For accurate and up-to-date information, it's recommended to check with the official channels of both Cartesi and Akash Network or consult with relevant parties involved in the integration.

Are there any reasons why bringing cartesi smart contracts to akash/ cosmos ecosystem may not work?

While integrating Cartesi smart contracts with the Akash/Cosmos ecosystem holds potential benefits, challenges could arise. Compatibility issues, differing consensus mechanisms, or technical complexities might pose obstacles. Additionally, community acceptance, governance alignment, and potential conflicts in development priorities could be factors impacting successful integration. Thorough technical assessments, collaboration, and community engagement are essential to address and overcome these challenges for a successful implementation.

I asked in cartesi and akash discords but from what I can tell no one else is working on this.

The project is open source

How long would it take to bring cartesi smart contracts to cosmos/ akash ecosystem?

The time required to bring Cartesi smart contracts to the Cosmos/Akash ecosystem depends on various factors, including the complexity of integration, development resources, and the collaboration between the teams involved. It's challenging to provide a specific timeframe without detailed information on the project scope and the speed of development progress. Close communication between Cartesi and Akash teams, along with community involvement, will play crucial roles in determining the timeline for a successful integration.

**Comment by hewi333** [comment]
Hi @31trainman,

Are you looking for a specific action or support on this or just looking to call it out as a potential item to look into?

**Reply by 31trainman** [reply]
I'm looking to try and get funding as well as support to make this happen.

**Reply by hewi333** [reply]
I'd check out best practices for submitting governance proposals [here.](https://github.com/orgs/akash-network/discussions/170)

Items like clearly defined problem, scope of work, timelines, dollar amounts, expected goals, etc. will all help you garner support from the community.

**Reply by 31trainman** [reply]
Thanks. I'll be sure to do that as I'm gonna guess this proposal is missing that information?

**Comment by anilmurty** [comment]
Comments from Steering Committee discusion:

1. Security concerns with enabling Cartesi smart contracts
2. We can potentially solve the use case without smart contracts

As a side note: The folks at Agoric have also floated the idea of smart contracts but we realized that this is something that can be achieved with Inter-Chain Accounts (ICA)

**Reply by 31trainman** [reply]
What's the security concerns exactly? I need these smart contracts because they can run Linux code and that's super useful.

**Reply by troian** [reply]
this discussion lacks:
- links to documentation about Cartesi smart contracts
- what tasks/issue this smart contracts implementation intends to solve and how integrating it to the network benefits it's users
- comparison with other existing known smart contract implementations like CosmoWasm

>  I need these smart contracts because they can run Linux code and that's super useful.
this is poorly formatted argument that does not hold.

anyways, at this moment Akash Network does not consider adding support for smart contracts, for various reasons, including, but not limited to:
- lack of use-cases
- majority (if not all) previously discussed use-cases ended up to be done with simple state change.

considering all above, this discussion will be closed and locked due considerable lack of information.
Feel free to open new discussion, however, to prevent future miss-understanding please, consider include as much information and details as possible.
